vault/resource_pki_secret_backend_intermediate_cert_request.go

```diff
@@ -262,15 +265,19 @@ func pkiSecretBackendIntermediateCertRequestCreate(ctx context.Context, d *schem
 
 	// add multi-issuer write API fields if supported
 	isIssuerAPISupported := provider.IsAPISupported(meta, provider.VaultVersion111)
+	isKeyBeingGenerated := !(intermediateType == keyTypeKMS || intermediateType == consts.FieldExisting)
```

Comment:
Sorry @vinay-gopalan, I thought about this last night and this is slightly incorrect. 

The `kms` exported type technically creates a key within PKI (just not key material, that is stored in the HSM).  So we need to guard `consts.FieldKeyType` and `consts.FieldKeyBits` against being used for `kms` and `existing` types but `consts.FieldKeyName` should be available to the `kms` type...

Reply:
Thanks @stevendpclark! I made the necessary updates and also added some better test coverage for the root and intermediate cert resources.